servers/list-api/src/resolvers/mutation.ts

```diff
@@ -421,3 +490,70 @@ export async function exportList(
   await exportListEvent(requestId, context.eventContext);
   return requestId;
 }
+
+/**
+ * Get a presigned url for uploading to a specific file key.
+ * Expires 5 minutes after grant.
+ * Requires PUT request to upload.
+ */
+export async function importUploadUrl(
+  _,
+  { importType }: { importType: keyof ImportMapping },
+  context: IContext,
+): Promise<
+  { url: string; ttl: number } | { message: string; refreshInHours: number }
+> {
+  // Once per day, use the date as key
+  // Hard-coded to a zipfile for now (omnivore) - TODO
+  const filename = `${new Date().toISOString().split('T')[0]}.zip`;
+  const fileKey = path.join(context.userId, importType, filename);
+
+  const s3: NodeJsClient<S3Client> = new S3Client({
```

Comment:
NodeJsClient - this is some new syntactic sugar 👁️

Reply:
yeah, it's because otherwise the types aren't narrowed for s3 (so you don't get the outputs properly, the streams aren't node streams, etc.)